core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -386,6 +387,18 @@ class SparkContext(config: SparkConf) extends Logging {
     Utils.setLogLevel(Level.toLevel(upperCased))
   }
 
+  // Local storage for transferred artifacts through Spark Connect.
+  private val _sparkConnectArtifactDirectory: File = Utils.createTempDir("artifacts")
+
+  /**
+   * :: Experimental ::
+   * Returns the directory that stores artifacts transferred through Spark Connect.
+   *
+   * @since 3.4.0
+   */
+  @Experimental
```

Comment:
Is this a user-facing API?

Reply:
This is primarily for access from the `connect` package but also carries a side benefit of allowing users to inspect the artifact directory. This is a temporary step for now (to pass information to the upper SQL layer) but it may be removed in the longer term (if we switch to passing information down to core from the SQL layer).

No requirement for making it user-facing at the moment so any suggestions to limit scope (while still being able to access it from the `connect` package) would be welcome.

Reply:
I don't think we should make it user-facing. If the accessor has to be public, we could mark it with `@Private`.

Reply:
Sounds good 👍